src/main/java/org/opensearch/ad/rest/handler/ModelValidationActionHandler.java

```diff
@@ -0,0 +1,718 @@
+     * Bucket aggregation with different interval lengths are executed one by one to check if the data is dense enough
+     * We only need to execute the next query if the previous one led to data that is too sparse.
+     */
+    class DetectorIntervalRecommendationListener implements ActionListener<SearchResponse> {
+        private final ActionListener<IntervalTimeConfiguration> intervalListener;
+        SearchSourceBuilder searchSourceBuilder;
+        IntervalTimeConfiguration detectorInterval;
+        private final long expirationEpochMs;
+        private final long latestTime;
+
+        DetectorIntervalRecommendationListener(
+            ActionListener<IntervalTimeConfiguration> intervalListener,
+            SearchSourceBuilder searchSourceBuilder,
+            IntervalTimeConfiguration detectorInterval,
+            long expirationEpochMs,
+            long latestTime
+        ) {
+            this.intervalListener = intervalListener;
+            this.searchSourceBuilder = searchSourceBuilder;
+            this.detectorInterval = detectorInterval;
+            this.expirationEpochMs = expirationEpochMs;
+            this.latestTime = latestTime;
+        }
+
+        private AggregationBuilder getNewAggregationBuilder(long newInterval) {
+            return AggregationBuilders
+                .dateHistogram("agg")
+                .field(anomalyDetector.getTimeField())
+                .minDocCount(0)
+                .hardBounds(getTimeRangeBounds(latestTime, new IntervalTimeConfiguration(newInterval, ChronoUnit.MINUTES)))
+                .fixedInterval(DateHistogramInterval.minutes((int) newInterval));
+        }
+
+        private long convertIntervalToMinutes(IntervalTimeConfiguration interval) {
+            long currentInterval = interval.getInterval();
+            if (interval.getUnit() == ChronoUnit.MILLIS) {
+                currentInterval /= 60000;
+            }
+            return currentInterval;
+        }
+
+        @Override
+        public void onResponse(SearchResponse response) {
+            try {
+                Histogram aggregate = checkBucketResultErrors(response);
+                if (aggregate == null) {
+                    return;
+                }
+
+                double fullBucketRate = processBucketAggregationResults(aggregate);
+                long newInterval = (long) Math.ceil(convertIntervalToMinutes(detectorInterval) + 1);
+                // If rate is below success minimum then call search again.
+                if (fullBucketRate > INTERVAL_BUCKET_MINIMUM_SUCCESS_RATE) {
+                    intervalListener.onResponse(this.detectorInterval);
+                } else if (expirationEpochMs < clock.millis()) {
+                    listener
+                        .onFailure(
+                            new AnomalyDetectionException(
+                                "Timed out getting interval recommendation. Please continue with detector creation."
```

Comment:
Second sentence assumes it is ok. Suggest to remove it or change to something like "Please try again"

Reply:
I could potentially remove this, right now we aren't targeting having a manual validation process for them to try again but more of just a message that they can continue. I might change this to a generic message that shows `"Model validation experienced issues completing."`

Reply:
Removing timeout clock completely as it doesn't seem to ever get even close to it

Reply:
Changed to generic model validation exception for now